modules/ml/src/main/java/org/apache/ignite/ml/naivebayes/gaussian/GaussianNaiveBayesTrainer.java

```diff
@@ -41,17 +43,10 @@
     /** Sets equivalent probability for all classes. */
     private boolean equiprobableClasses;
 
-    /**
-     * Trains model based on the specified data.
-     *
-     * @param datasetBuilder Dataset builder.
-     * @param featureExtractor Feature extractor.
-     * @param lbExtractor Label extractor.
-     * @return Model.
-     */
+    /** {@inheritDoc} */
     @Override public <K, V> GaussianNaiveBayesModel fit(DatasetBuilder<K, V> datasetBuilder,
-        IgniteBiFunction<K, V, Vector> featureExtractor, IgniteBiFunction<K, V, Double> lbExtractor) {
-        return updateModel(null, datasetBuilder, featureExtractor, lbExtractor);
+        IgniteBiFunction<K, V, SimpleLabeledVector<Double>> extractor) {
```

Comment:
Does it make sense to define a dedicated class like `FeatureLabelExtractor` with methods like `extract()`, `extractFeatures()` and `extractLabel()`.